src/data/CHDatasource.ts

```diff
@@ -1,29 +1,36 @@
 import {
+  ArrayDataFrame,
   DataFrame,
   DataQueryRequest,
   DataQueryResponse,
   DataSourceInstanceSettings,
+  MetricFindValue,
   ScopedVars,
   vectorator,
 } from '@grafana/data';
-import { DataSourceWithBackend, getTemplateSrv } from '@grafana/runtime';
+import { DataSourceWithBackend, getTemplateSrv, TemplateSrv } from '@grafana/runtime';
 import { CHConfig, CHQuery } from '../types';
+import AdHocManager from './adHocManager';
 
 export class Datasource extends DataSourceWithBackend<CHQuery, CHConfig> {
   // This enables default annotation support for 7.2+
   annotations = {};
   settings: DataSourceInstanceSettings<CHConfig>;
+  templateSrv: TemplateSrv;
+  adHocManager: AdHocManager;
 
   constructor(instanceSettings: DataSourceInstanceSettings<CHConfig>) {
     super(instanceSettings);
     this.settings = instanceSettings;
+    this.templateSrv = getTemplateSrv();
+    this.adHocManager = new AdHocManager();
```

Comment:
not sure why we need this.  can we just import the function and use it where its needed?